File: frontend/src/types/stats.ts
/**
 * TypeScript types for dashboard statistics
 */

export interface OverviewStats {
  total_articles: number;
  total_claims: number;
  total_investigations: number;
  last_ingestion: string | null;
}

export interface VerdictDistribution {
  true: number;
  mostly_true: number;
  mixed: number;
  mostly_false: number;
  false: number;
  unverifiable: number;
}

export interface RecentActivity {
  time_range: string;
  new_articles: number;
  new_claims: number;
  new_investigations: number;
}

export interface QualityMetrics {
  avg_confidence: number;
  avg_propaganda_score: number;
  avg_source_reliability: number;
}

export interface ProcessingQueue {
  pending_articles: number;
  processing_articles: number;
  pending_claims: number;
  checking_claims: number;
}

export interface TrendingClaim {
  claim_text: string;
  verdict: string;
  confidence: number;
  article_count: number;
}

export interface PropagandaTechnique {
  technique: string;
  count: number;
}

export interface ProblematicSource {
  source_name: string;
  propaganda_score: number;
  article_count: number;
}

export interface PropagandaAnalysis {
  top_techniques: PropagandaTechnique[];
  problematic_sources: ProblematicSource[];
}

export interface DashboardStats {
  overview: OverviewStats;
  verdict_distribution: VerdictDistribution;
  recent_activity: RecentActivity;
  quality_metrics: QualityMetrics;
  processing_queue: ProcessingQueue;
  trending_claims: TrendingClaim[];
  propaganda_analysis: PropagandaAnalysis;
}

File: backend/app/schemas/stats.py
"""Pydantic schemas for dashboard statistics."""
from typing import List, Optional
from pydantic import BaseModel


class OverviewStats(BaseModel):
    """Overview statistics model."""
    total_articles: int
    total_claims: int
    total_investigations: int
    last_ingestion: Optional[str] = None


class VerdictDistribution(BaseModel):
    """Verdict distribution model."""
    true: int = 0
    mostly_true: int = 0
    mixed: int = 0
    mostly_false: int = 0
    false: int = 0
    unverifiable: int = 0


class RecentActivity(BaseModel):
    """Recent activity model."""
    time_range: str
    new_articles: int
    new_claims: int
    new_investigations: int


class QualityMetrics(BaseModel):
    """Quality metrics model."""
    avg_confidence: float
    avg_propaganda_score: float
    avg_source_reliability: float


class ProcessingQueue(BaseModel):
    """Processing queue status model."""
    pending_articles: int
    processing_articles: int
    pending_claims: int
    checking_claims: int


class TrendingClaim(BaseModel):
    """Trending claim model."""
    claim_text: str
    verdict: str
    confidence: float
    article_count: int


class PropagandaTechnique(BaseModel):
    """Propaganda technique model."""
    technique: str
    count: int


class ProblematicSource(BaseModel):
    """Problematic source model."""
    source_name: str
    propaganda_score: float
    article_count: int


class PropagandaAnalysis(BaseModel):
    """Propaganda analysis model."""
    top_techniques: List[PropagandaTechnique]
    problematic_sources: List[ProblematicSource]


class DashboardStatsResponse(BaseModel):
    """Complete dashboard statistics response model."""
    overview: OverviewStats
    verdict_distribution: VerdictDistribution
    recent_activity: RecentActivity
    quality_metrics: QualityMetrics
    processing_queue: ProcessingQueue
    trending_claims: List[TrendingClaim]
    propaganda_analysis: PropagandaAnalysis
